pkg/platform/automate/share.go

```
@@ -0,0 +1,269 @@
+	}
+
+	for _, deployment := range deployments {
+		microservice := ConvertObjectMetaToMicroservice(deployment.GetObjectMeta())
+		microservices = append(microservices, microservice)
+	}
+	return microservices, nil
+}
+
+func GetNamespaces(ctx context.Context, client kubernetes.Interface) []corev1.Namespace {
+	namespacesList, err := client.CoreV1().Namespaces().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		panic(err.Error())
+	}
+	return namespacesList.Items
+}
+
+func IsApplicationNamespace(namespace corev1.Namespace) bool {
+	if !strings.HasPrefix(namespace.GetName(), "application-") {
+		return false
+	}
+	if _, hasAnnotation := namespace.Annotations["dolittle.io/tenant-id"]; !hasAnnotation {
+		return false
+	}
+	if _, hasAnnotation := namespace.Annotations["dolittle.io/application-id"]; !hasAnnotation {
+		return false
+	}
+	if _, hasLabel := namespace.Labels["tenant"]; !hasLabel {
+		return false
+	}
+	if _, hasLabel := namespace.Labels["application"]; !hasLabel {
+		return false
+	}
+
+	return true
+}
+
+func GetMicroserviceDirectory(rootFolder string, configMap corev1.ConfigMap) string {
+	labels := configMap.GetObjectMeta().GetLabels()
+	customer := labels["tenant"]
+	application := labels["application"]
+	environment := labels["environment"]
+	microservice := labels["microservice"]
+
+	return filepath.Join(rootFolder,
+		"Source",
+		"V3",
+		"Kubernetes",
+		"Customers",
+		customer,
+		application,
+		environment,
+		microservice,
+	)
+}
+
+func UpdateConfigMap(ctx context.Context, client kubernetes.Interface, logContext logrus.FieldLogger, configMap corev1.ConfigMap, dryRun bool) error {
+	microservice := ConvertObjectMetaToMicroservice(configMap.GetObjectMeta())
+	platform := configK8s.NewMicroserviceConfigmapPlatformData(microservice)
+	b, _ := json.MarshalIndent(platform, "", "  ")
+	platformJSON := string(b)
+
+	if configMap.Data == nil {
+		// TODO this is a sign it might not be using a runtime, maybe we skip
```

Should we warn?
I cant decide if we should return an error here.
Perhaps later, when we do know the microservice kind, we can add logic to handle if it should do what it currently does, or return an error.

Ie simple == includes runtime and should error

Hmm I just realised, we can have "simple" and not use the runtime.... we might have to return to this.

---

I started to think first that I don't think it's the `UpdateConfigMap()` functions job to check and make assumptions about what kind of configmap it's supposed to update. But then I realised that this function doesn't actually just update configmaps as a generic thing, but rather only adds the specific `platform.json` part into it. In that case either this function (which could maybe be called `AddMicroserviceConfigmapPlatformData()` or something equally ugly and verbose) or it's caller should make sure that the given configmap is a dolittle configmap.

Currently I don't think adding some new data to the dolittle configmaps (even the ones that don't use the Runtime) would create any problems except it would add noise to the configmaps. But you could make the same argument for adding the `platform.json` data into microservices that aren't going to use it...

And as you said, we don't really use the `microservice-kind` annotation in a defined way currently so I don't think you can draw too many conclusions from it.

So in the end I'm fine if we error there or write a warning and skip the update.

---

We should rename this function to make it clearer what it is doing.

microservice-kind, is going to be added https://app.asana.com/0/1201325052247030/1201451644838466/f.

---

I've renamed it to `AddDataToConfigMap()`

---

Nice :) I am excited by this func.